### PR TITLE
fix: url input field overflows when update profile & editing event

### LINKED ISSUE #13786: [CAL-3197] The Url input field overflows when updating profile
### Issue Summary

The URL input field overflows when the URL is very long, breaking the UI.

### Steps to Reproduce

1. Go to My setting
2. Clicked on Profile
3. Update the input field of URL

### Actual Results

1. The URL field overflows horizontally breaking the UI
2. Check in the popup and the also in the events edit section

### Expected Results

1. The URL field should accommodate long URL strings.

### Evidence

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3197](https://linear.app/calcom/issue/CAL-3197/the-url-input-field-overflows-when-updating-profile)</sub>

### PR DESCRIPTION
**Pull request by rajats-dev**
## What does this PR do?

This PR fixes the issue of breaking the UI when url input field overflows.

Fixes #13786

Screenshot 2024-03-08 140048
Screenshot 2024-03-08 150137

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Steps to Reproduce
- Go to My setting
- Clicked on Profile
- Update the input field of URL
- Go to the event & edit the event.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by PeerRich** [comment]
i believe this has already been fixed, @Udit-takkar knows more

**Comment by rajats-dev** [comment]
> i believe this has already been fixed, @Udit-takkar knows more

I just check out that PR, it was not complete and issue was still there.

File: apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx [hunk 1]
```diff
@@ -157,20 +157,22 @@ const UsernameTextfield = (props: ICustomUsernameProps & Partial<React.Component
...
-              <div className="bg-subtle flex w-full flex-wrap gap-6 rounded-sm px-2 py-3 text-sm">
+              <div className="bg-subtle flex w-full flex-wrap justify-between gap-6 rounded-sm  px-4 py-3 text-sm">
                 <div>
                   <p className="text-subtle">{t("current_username")}</p>
-                  <p className="text-emphasis mt-1 break-all" data-testid="current-username">
-                    {currentUsername}
+                  <p
+                    className="text-emphasis min-h-5 mt-1 max-h-12 overflow-auto break-all"
+                    data-testid="current-username">
+                    {currentUsername?.substring(0, 200)}
```
Comment (Udit-takkar) [inline apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx:166]: we want to display the full username

File: apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx [hunk 2]
```diff
@@ -157,20 +157,22 @@ const UsernameTextfield = (props: ICustomUsernameProps & Partial<React.Component
...
                   </p>
                 </div>
                 <div>
                   <p className="text-subtle" data-testid="new-username">
                     {t("new_username")}
                   </p>
 
-                  <p className="text-emphasis min-h-5 mt-1 max-h-12 overflow-y-auto break-all">
-                    {inputUsernameValue}
+                  <p className="text-emphasis min-h-5 mt-1 max-h-12 overflow-auto break-all">
```
[the remark this answers is not in the document]
Comment (rajats-dev) [reply apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx:174]: Overflow-y-auto will also work.

**Review by Udit-takkar — DISMISSED** [review]
Screenshot 2024-03-13 at 3 20 21 PM

the tick icon at the end of input is overlapping with text

File: apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Udit-takkar) [reply apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx:174]: we don't want horizontal scroll

**Comment by rajats-dev** [comment]
> Screenshot 2024-03-13 at 3 20 21 PM
> the tick icon at the end of input is overlapping with text

I did not make any changes to this one, it was not mentioned in the issue.

File: apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (rajats-dev) [reply apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx:174]: so what do yo suggest?

File: apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx [hunk 3]
```diff
@@ -135,8 +135,8 @@ const UsernameTextfield = (props: ICustomUsernameProps & Partial<React.Component
             {...rest}
           />
           {currentUsername !== inputUsernameValue && (
-            <div className="absolute right-[2px] top-6 flex flex-row">
-              <span className={classNames("mx-2 py-3.5")}>
+            <div className="absolute right-[2px] top-6 flex h-7 flex-row">
+              <span className={classNames("mx-0 bg-white p-3")}>
```
Comment (sean-brydon) [inline apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx:139]: I dont think we can use bg-white here as we need this to work in dark mode also. Off the top of my head `bg-default` should work. 

Thanks

**Review by sean-brydon — CHANGES_REQUESTED** [review]

File: apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx (hunk 3 above)
Comment (rajats-dev) [reply apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx:139]: Actually I applied dark mode with chrome extension it is working.

**Review by sean-brydon — DISMISSED** [review]
CleanShot 2024-03-19 at 08 53 54@2x
LGTM

**Review by Udit-takkar — DISMISSED** [review]

**Comment by CarinaWolli** [comment]
@rajats-dev there are type errors, can you take a look?

**Comment by rajats-dev** [comment]
> @rajats-dev there are type errors, can you take a look?

@Udit-takkar added a tooltip attribute. I think those are having type error.

**Review by Udit-takkar — APPROVED** [review]

**Pull request merged**